qubesmanager/backup_utils.py

```diff
@@ -20,90 +19,98 @@
-    if len(untrusted_path) == 0:
-        return None
-    if path_re.match(untrusted_path):
-        assert '../' not in untrusted_path
-        assert '\0' not in untrusted_path
-        return untrusted_path.strip()
-    else:
-        return None
-
-def select_path_button_clicked(dialog, select_file = False):
+
+def select_path_button_clicked(dialog, select_file=False, read_only=False):
     backup_location = str(dialog.dir_line_edit.text())
-    file_dialog = QFileDialog()
+    file_dialog = QtGui.QFileDialog()
     file_dialog.setReadOnly(True)
 
-    if select_file:
-        file_dialog_function = file_dialog.getOpenFileName
-    else:
-        file_dialog_function = file_dialog.getExistingDirectory
-
-    new_appvm = None
     new_path = None
-    if dialog.appvm_combobox.currentIndex() != 0:   #An existing appvm chosen
-        new_appvm = str(dialog.appvm_combobox.currentText())
-        vm = dialog.qvm_collection.get_vm_by_name(new_appvm)
-        if vm:
-            new_path = get_path_for_vm(vm, "qubes.SelectFile" if select_file
-                    else "qubes.SelectDirectory")
-    else:
-        new_path = file_dialog_function(dialog,
-            dialog.tr("Select backup location."),
-            backup_location if backup_location else '/')
-
-    if new_path != None:
-        if os.path.basename(new_path) == 'qubes.xml':
-            backup_location = os.path.dirname(new_path)
+
+    new_appvm = str(dialog.appvm_combobox.currentText())
+    vm = dialog.qvm_collection.domains[new_appvm]
+    try:
+        new_path = utils.get_path_from_vm(
+            vm,
+            "qubes.SelectFile" if select_file
+            else "qubes.SelectDirectory")
+    except subprocess.CalledProcessError:
+        if not read_only:
+            QtGui.QMessageBox.warning(
+                None,
+                dialog.tr("Nothing selected!"),
+                dialog.tr("No file or directory selected."))
         else:
-            backup_location = new_path
-        dialog.dir_line_edit.setText(backup_location)
+            return
+
+    if new_path and not read_only:
+        dialog.dir_line_edit.setText(new_path)
+
+    if new_path and backup_location and not read_only:
+        dialog.select_dir_page.emit(QtCore.SIGNAL("completeChanged()"))
+
+
+def load_backup_profile(use_temp=False):
+
+    path = get_profile_path(use_temp)
+
+    with open(path) as profile_file:
+        profile_data = yaml.safe_load(profile_file)
+    return profile_data
+
+
+def write_backup_profile(args, use_temp=False):
+
+    acceptable_fields = ['include', 'passphrase_text', 'compression',
+                         'destination_vm', 'destination_path']
+
+    profile_data = {key: value for key, value in args.items()
+                    if key in acceptable_fields}
+
+    path = get_profile_path(use_temp)
+
+    with open(path, 'w') as profile_file:
+        yaml.safe_dump(profile_data, profile_file)
+
+
+def get_profile_name(use_temp):
```

Comment:
Logically this (and `get_profile_path`) should be before `load_backup_profile` and `write_backup_profile`.